### PR TITLE
sourcehut: update all components

### PR DESCRIPTION
**Pull request by christoph-heiss**
## Description of changes

This is a new round of updates across the sourcehut stack. Keeping close to upstream should help us all around, as big version upgrades can be painful.
Pretty minor version upgrades for most packages, nothing should be broken (or at least; not more than before).

It has also been tested on top of the incoming, much more expansive testsuite in #271324.

Two more things to note:
* I've added myself as maintainer for anything sourcehut, since I already spend way to much time on this.
* Refactored the `todo`, `man`, `lists`, `hub` and `hg` packages, such that they are all now pretty similar. I would then combine them using some common expression and factor out most of the now-common stuff between all of them as a next step.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/version-management/sourcehut/core.nix [hunk 1]
```diff
@@ -37,7 +37,7 @@ buildPythonPackage rec {
     owner = "~sircmpwn";
     repo = "core.sr.ht";
     rev = version;
-    sha256 = "sha256-T9yewweqnWL3IW5PHGyAcsIWCGn1ayK2rwrHVukYpgE=";
+    sha256 = "sha256-YIoKOiTi/9X4bSiG+GvnwzvKYhbfywrv/dTjxaJOOTQ=";
```
Comment (mfrw) [inline pkgs/applications/version-management/sourcehut/core.nix:40]:
```suggestion
    hash = "sha256-YIoKOiTi/9X4bSiG+GvnwzvKYhbfywrv/dTjxaJOOTQ=";
```

File: pkgs/applications/version-management/sourcehut/git.nix [hunk 2]
```diff
@@ -13,70 +13,70 @@
...
-  version = "0.84.2";
+  version = "0.85.7";
+  gqlgen = import ./fix-gqlgen-trimpath.nix { inherit unzip; };
 
   src = fetchFromSourcehut {
     owner = "~sircmpwn";
     repo = "git.sr.ht";
     rev = version;
-    sha256 = "sha256-sAkTsQlWtNDQ5vAhA2EeOvuJcj9A6AG8pgDyIKtr65s=";
+    sha256 = "sha256-jkESrrVE+0O2g64zzPOpqhl8DpvmosQvuF0s6Xd+lbM=";
```
Comment (mfrw) [inline pkgs/applications/version-management/sourcehut/git.nix:23]:
```suggestion
    hash = "sha256-jkESrrVE+0O2g64zzPOpqhl8DpvmosQvuF0s6Xd+lbM=";
```

File: pkgs/applications/version-management/sourcehut/hub.nix [hunk 3]
```diff
@@ -11,30 +11,34 @@
...
+let
+  version = "0.17.5";
+  gqlgen = import ./fix-gqlgen-trimpath.nix { inherit unzip; gqlgenVersion = "0.17.41"; };
 
   src = fetchFromSourcehut {
     owner = "~sircmpwn";
     repo = "hub.sr.ht";
     rev = version;
-    sha256 = "sha256-A+lvRsPz5EBnM0gB4PJuxSMpELZTrK14ORxDbTKPXWg=";
+    sha256 = "sha256-GbBxK3XE+Y6Jiap0Nxa8vk4Kv6IbcdSi4NN59AeKwjA=";
```
Comment (mfrw) [inline pkgs/applications/version-management/sourcehut/hub.nix:22]:
```suggestion
    hash = "sha256-GbBxK3XE+Y6Jiap0Nxa8vk4Kv6IbcdSi4NN59AeKwjA=";
```

File: pkgs/applications/version-management/sourcehut/lists.nix [hunk 4]
```diff
@@ -14,26 +14,30 @@
...
+let
+  version = "0.57.14";
+  gqlgen = import ./fix-gqlgen-trimpath.nix { inherit unzip; gqlgenVersion = "0.17.36"; };
 
   src = fetchFromSourcehut {
     owner = "~sircmpwn";
     repo = "lists.sr.ht";
     rev = version;
-    sha256 = "sha256-nQZRSTAyTWxcPHrRVCZ5TgcrNgrlxBFc1vRds0cQwA0=";
+    sha256 = "sha256-rzOxlat7Lbgt0Wl6vvnAC+fS3MynFVKFvVdIdxgA5e0=";
```
Comment (mfrw) [inline pkgs/applications/version-management/sourcehut/lists.nix:25]:
```suggestion
    hash = "sha256-rzOxlat7Lbgt0Wl6vvnAC+fS3MynFVKFvVdIdxgA5e0=";
```

File: pkgs/applications/version-management/sourcehut/man.nix [hunk 5]
```diff
@@ -11,30 +11,34 @@
...
+let
+  version = "0.16.3";
+  gqlgen = import ./fix-gqlgen-trimpath.nix { inherit unzip; gqlgenVersion = "0.17.36"; };
 
   src = fetchFromSourcehut {
     owner = "~sircmpwn";
     repo = "man.sr.ht";
     rev = version;
-    sha256 = "sha256-94G9/Kzt1gaQ2CaXtsJYCB6W5OTdn27XhVdpNJ9a5cE=";
+    sha256 = "sha256-o1A3LmwH6WgpFqjKyL3UTru9q7TgKdOdbKZfJHR6fCA=";
```
Comment (mfrw) [inline pkgs/applications/version-management/sourcehut/man.nix:22]:
```suggestion
    hash = "sha256-o1A3LmwH6WgpFqjKyL3UTru9q7TgKdOdbKZfJHR6fCA=";
```

File: pkgs/applications/version-management/sourcehut/pages.nix [hunk 6]
```diff
@@ -6,21 +6,21 @@
...
   pname = "pagessrht";
-  version = "0.13.0";
+  version = "0.15.4";
 
   src = fetchFromSourcehut {
     owner = "~sircmpwn";
     repo = "pages.sr.ht";
     rev = version;
-    sha256 = "sha256-vUN6c6cyhcLI8bKrFYKoxlBQ29VS/bowpSfBRmi47wg=";
+    sha256 = "sha256-3kdQVIL7xaIPu2elxj1k+4/y75bd+OKP5+VPSniF7w8=";
```
Comment (mfrw) [inline pkgs/applications/version-management/sourcehut/pages.nix:15]:
```suggestion
    hash = "sha256-3kdQVIL7xaIPu2elxj1k+4/y75bd+OKP5+VPSniF7w8=";
```

File: pkgs/applications/version-management/sourcehut/todo.nix [hunk 7]
```diff
@@ -12,31 +12,35 @@
...
+let
+  version = "0.75.6";
+  gqlgen = import ./fix-gqlgen-trimpath.nix { inherit unzip; gqlgenVersion = "0.17.36"; };
 
   src = fetchFromSourcehut {
     owner = "~sircmpwn";
     repo = "todo.sr.ht";
     rev = version;
-    sha256 = "sha256-j12pCGfKf6+9R8NOBIrH2V4OuSMuncU6S1AMWFVoHts=";
+    sha256 = "sha256-BPJ1M9dX+xNIw++VZ0Si/rjnfI9BY95TE2o+u7JRVAU=";
```
Comment (mfrw) [inline pkgs/applications/version-management/sourcehut/todo.nix:23]:
```suggestion
    hash = "sha256-BPJ1M9dX+xNIw++VZ0Si/rjnfI9BY95TE2o+u7JRVAU=";
```

**Review by mfrw — COMMENTED** [review]
A few cosmetic suggestions: `s/sha256/hash/g` :)

File: pkgs/applications/version-management/sourcehut/core.nix (hunk 1 above)
Comment (christoph-heiss) [reply pkgs/applications/version-management/sourcehut/core.nix]: Thanks for the review! Fixed all instances :+1:

**Comment by mfrw** [comment]
Result of `nixpkgs-review pr 286993` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>21 packages built:</summary>
  <ul>
    <li>sourcehut.buildsrht</li>
    <li>sourcehut.buildsrht.dist</li>
    <li>sourcehut.coresrht</li>
    <li>sourcehut.coresrht.dist</li>
    <li>sourcehut.gitsrht</li>
    <li>sourcehut.gitsrht.dist</li>
    <li>sourcehut.hgsrht</li>
    <li>sourcehut.hgsrht.dist</li>
    <li>sourcehut.hubsrht</li>
    <li>sourcehut.hubsrht.dist</li>
    <li>sourcehut.listssrht</li>
    <li>sourcehut.listssrht.dist</li>
    <li>sourcehut.mansrht</li>
    <li>sourcehut.mansrht.dist</li>
    <li>sourcehut.metasrht</li>
    <li>sourcehut.metasrht.dist</li>
    <li>sourcehut.pagessrht</li>
    <li>sourcehut.pastesrht</li>
    <li>sourcehut.pastesrht.dist</li>
    <li>sourcehut.todosrht</li>
    <li>sourcehut.todosrht.dist</li>
  </ul>
</details>

**Review by nessdoor — APPROVED** [review]
Thank you for taking on maintainership and keeping track of upstream changes! All those `gqlgen` dependencies always make me so nervous...

+1 for the intention of factoring common parts more rationally, all these versions and hashes scattered around make version bumps so very tedious.

I also tested compilation on an AArch64 box; results follow.

Result of `nixpkgs-review pr 286993` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
  </ul>
</details>
<details>
  <summary>21 packages built:</summary>
  <ul>
    <li>sourcehut.buildsrht</li>
    <li>sourcehut.buildsrht.dist</li>
    <li>sourcehut.coresrht</li>
    <li>sourcehut.coresrht.dist</li>
    <li>sourcehut.gitsrht</li>
    <li>sourcehut.gitsrht.dist</li>
    <li>sourcehut.hgsrht</li>
    <li>sourcehut.hgsrht.dist</li>
    <li>sourcehut.hubsrht</li>
    <li>sourcehut.hubsrht.dist</li>
    <li>sourcehut.listssrht</li>
    <li>sourcehut.listssrht.dist</li>
    <li>sourcehut.mansrht</li>
    <li>sourcehut.mansrht.dist</li>
    <li>sourcehut.metasrht</li>
    <li>sourcehut.metasrht.dist</li>
    <li>sourcehut.pagessrht</li>
    <li>sourcehut.pastesrht</li>
    <li>sourcehut.pastesrht.dist</li>
    <li>sourcehut.todosrht</li>
    <li>sourcehut.todosrht.dist</li>
  </ul>
</details>

**Review by mfrw — APPROVED** [review]
LGTM :)
Thanks

**Review by tomberek — APPROVED** [review]

**Pull request merged**